src/game_main.hpp

```diff
@@ -18,7 +18,7 @@
 
 #include "common/command_line_options.hpp"
 
-
+#include "emscripten.h"
```

Comment:
I think this can go into the `.cpp`, it shouldn't be needed in the header file. Also, we would need some conditional compilation here, so that it's only included when building with Emscripten

Reply:
Another option would be to change things so that we could set the mainloop callback from `emscripten_main.cpp`, to keep `game_main.cpp` unaware of Emscripten. Since the former would only be built when building with Emscripten, it would then be ok to unconditionally include the Emscripten header.

The idea would be to add two more functions to `game_main.hpp`:

```cpp
std::unique_ptr<Game> createGame();
void runOneFrame(Game* game);
```

In the regular `main.cpp`, we would then continue calling `gameMain` as before, but in `emscripten_main.cpp`, we would do something like:

```cpp
void callback(void* pData) {
  rigel::runOneFrame(static_cast<rigel::Game*>(pData));
}

int main() {
  auto pGame = rigel::createGame();
  emscripten_set_main_loop_arg(callback, pGame.get(), 0, true);
  return 0;
}
```

This way, the different code paths are nicely separated. And we could even add some more code later to allow passing some command line options via URL parameters or something, if we wanted to.